lib/sensors/gyroscope.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "i2c_sensors.hpp"
+
+#include <cstdint>
+#include <optional>
+
+#include <core/logger.hpp>
```

Comment:
`#include <core/types.hpp>` This has been forward included from "i2c_sensors.hpp" but explicitly include it since you use the `Axis` enum class here and we want to keep the codebase as modular as possible.